src/v/iceberg/json_writer.h

```diff
@@ -0,0 +1,27 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "json/chunked_buffer.h"
+#include "json/iobuf_writer.h"
+
+namespace iceberg {
+
+using json_writer = json::iobuf_writer<json::chunked_buffer>;
+
+template<typename T>
+ss::sstring to_json_str(const T& t) {
+    json::chunked_buffer buf;
+    iceberg::json_writer w(buf);
+    rjson_serialize(w, t);
+    auto p = iobuf_parser(std::move(buf).as_iobuf());
+    return p.read_string(p.bytes_left());
```

Comment:
The only uses of this (outside of tests) are into a `std::string`, so I'd be tempted to avoid the intermediate `iobuf` and `ss::sstring`, by using `json::StringBuffer` or creating a `json::std_string_buffer`, assuming the types are small.

I guess the main benefit of using chunked is to reduce the number of intermediate oversize reallocations?

Oh, I see, the main benefit is not having to template `rjson_serialize`!

Unless we do something with Avro, `std::map<std::string, std::string>` is not a great interface for us, I guess.

Reply:
Yea... the Apache Avro lib is pretty rough... I think we'll eventually need to write our own seastar-friendly library to get out of the ugly interfaces